routing_model/src/scenario_tests.rs

```diff
@@ -825,6 +825,100 @@ mod dst_tests {
             &AssertState::default(),
         );
     }
+
+    #[test]
+    fn test_local_events_offline_online_again_for_different_nodes() {
+        run_test(
+            "Get local event node detected offline online again different nodes",
+            &intial_state_old_elders(),
+            &[
+                LocalEvent::NodeDetectedOffline(NODE_ELDER_130).to_event(),
+                LocalEvent::NodeDetectedBackOnline(NODE_ELDER_131).to_event(),
+            ],
+            &AssertState {
+                action_our_votes: vec![
+                    ParsecVote::Offline(NODE_ELDER_130),
+                    ParsecVote::BackOnline(NODE_ELDER_131),
+                ],
+                ..AssertState::default()
+            },
+        );
+    }
+
+    #[test]
+    fn test_local_event_offline_then_parsec_offline() {
+        run_test(
+            "Get local event node detected offline and then parsec consensus offline",
+            &intial_state_old_elders(),
+            &[ParsecVote::Offline(NODE_ELDER_130).to_event()],
+            &AssertState {
+                action_our_nodes: vec![NodeChange::Offline(NODE_ELDER_130)],
+                ..AssertState::default()
+            },
+        );
+    }
+
+    #[test]
+    fn test_local_event_offline_and_parsec_offline_then_check_elder() {
+        let initial_state = arrange_initial_state(
+            &intial_state_old_elders(),
+            &[ParsecVote::Offline(NODE_ELDER_130).to_event()],
+        );
+        let new_votes = vec![
+            ParsecVote::AddElderNode(YOUNG_ADULT_205),
```

Comment:
I think this will be different once you fixed the filter, because there should be a remove elder here